test/simple_connection_test.exs

```diff
@@ -118,14 +118,34 @@ defmodule SimpleConnectionTest do
       assert_receive {:connect, i1}
 
       :sys.suspend(context.conn)
-      task = Task.async(fn -> SC.call(context.conn, {:query, "SELECT 1"}) end)
+
+      :erlang.trace(:all, true, [:call])
+      match_spec = [{:_, [], [{:return_trace}]}]
+      :erlang.trace_pattern({SC, :_, :_}, match_spec, [:local])
+
+      task =
+        Task.async(fn ->
+          # Simulate a small delay, because it used to cause flaky test failures
+          # in the past.
+          # https://github.com/elixir-ecto/postgrex/actions/runs/4151824827/jobs/7182370012
+          Process.sleep(10)
```

Comment:
Do we need to sleep if we have the trace?

Reply:
I think yes as a regression. I added a comment for it too 👍

Reply:
In my tests tracing `SC.call` was enough. I triggered CI with just that and if it passes I think we're ok to merge that, otherwise yeah I can trace the query getting to the connection.